### ISSUE TITLE
Question: what to do with newly detected ads while vault is open?

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam2#50]
Adding them to the vault will cause a reload of all ads (so that packing can occur). The other option is ignoring them.

**Comment by mushon** [comment]
Right now repacking would be a problem as it would reshuffle the ad locations. So that option might be too disruptive. Option two is indeed to ignore it, but it would be a shame to miss the opportunity to visualize the realtimeness of the system. What if we simply stick the new ads on the outside margins of the ad blob? (I realize "simply" is not necessarily the right word to use here). Is this even a direction worth exploring?

**Comment by dhowe** [comment]
This is indeed the 3rd option, yet we'd have to come up with some sensible algorithm for this. For example, a long banner ad stuck to the side of an existing pack would look quite bad...

**Comment by mushon** [comment]
True. 
I guess the algorithm should compare all the options for locating the banner and chose the one that least extends the blob's outer radius. 

Easier said than coded…

**Comment by mushon** [comment]
@dhowe please let me know how would you want to move forward on this, it's up to you really because the 3 different options mean different implementations…

**Comment by dhowe** [comment]
I think we put this on hold for now, with the goal of implementing a simple algorithm for adding new ads without repacking. Meantime, however, do we want a refresh button or note (that refreshes the vault, thus including the newly found ads) ?

**Comment by mushon** [comment]
Maybe a note that updates the number of ads waiting to be added and allows button to add them in?

**Comment by dhowe** [comment]
works for me... though for now, 'adding them in' will mean a repack

**Comment by mneunomne** [comment]
@dhowe This refers to the *Packing algorithm* we were talking about, perhaps we can keep this in mind if we are planning a refactor of the vault. But maybe the suggestion raised here is not relevant anymore.

**Comment by dhowe** [comment]
I believe there is already an implementation of this feature in some form. Please test and report what you see below...

**Comment by mneunomne** [comment]
When a new Ad is detected, it is added by repacking. For now I see not other alternative since any changes in the grid now results in repacking. 

Unless we completely reconsider the algorithm for the vault visualisation, I would not change this behaviour.

**Comment by dhowe** [comment]
This could be quite annoying however if you are trying to view your vault and it continually repacks

Another options would be to simply show the number of ads added since the last repack and a button to do the repack

**Comment by mneunomne** [comment]
I think an "UPDATE" button following the pattern of other pages would do, with the information of number of new ads to be shown. 
![Screen Shot 2021-03-19 at 14 51 29](https://user-images.githubusercontent.com/4967860/111790726-cb410100-88c2-11eb-9f51-eda0c6963c13.png)

**Comment by dhowe** [comment]
makes sense -- as long as its clear these are new ads _since_ the vault was opened
